.pipelines/templates/create-cluster.yaml

```diff
@@ -13,24 +13,39 @@ jobs:
   - job: ${{ parameters.name }}
     displayName: Cluster - ${{ parameters.name }}
     steps:
-      - task: AzureCLI@1
-        inputs:
-          azureSubscription: $(AZURE_TEST_AGENT_SERVICE_CONNECTION)
-          scriptLocation: "inlineScript"
-          scriptType: "bash"
-          addSpnToEnvironment: true
-          inlineScript: |
-            set -e
-            echo "Check az version"
-            az version
-            if ${{ lower(contains(parameters.clusterType, 'dualstack')) }}
-            then
+      - ${{ if contains(parameters.clusterType, 'dualstack') }}:
+        - task: AzureCLI@1
+          inputs:
+            azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+            scriptLocation: "inlineScript"
+            scriptType: "bash"
+            addSpnToEnvironment: true
+            inlineScript: |
+              set -e
+              echo "Check az version"
+              az version
               echo "Install az cli extension preview"
               az extension add --name aks-preview
               az extension update --name aks-preview
-            fi
-            mkdir -p ~/.kube/
-            make -C ./hack/aks azcfg AZCLI=az REGION=${{ parameters.region }}
-            make -C ./hack/aks ${{ parameters.clusterType }} AZCLI=az REGION=${{ parameters.region }} SUB=$(SUB_AZURE_NETWORK_AGENT_TEST) CLUSTER=${{ parameters.clusterName }} K8S_VER=${{ parameters.k8sVersion }} VM_SIZE=${{ parameters.vmSize }} WINDOWS_OS_SKU=${{ parameters.windowsOsSku }} WINDOWS_VM_SKU=${{ parameters.vmSize }} WINDOWS_USERNAME=${WINDOWS_USERNAME} WINDOWS_PASSWORD=${WINDOWS_PASSWORD}
-            echo "Cluster successfully created"
-        displayName: Cluster - ${{ parameters.clusterType }}
+              mkdir -p ~/.kube/
+              make -C ./hack/aks azcfg AZCLI=az REGION=${{ parameters.region }}
+              make -C ./hack/aks ${{ parameters.clusterType }} AZCLI=az REGION=${{ parameters.region }} SUB=$(DUALSTCK_OVERLAY_SUB) CLUSTER=${{ parameters.clusterName }} K8S_VER=${{ parameters.k8sVersion }} VM_SIZE=${{ parameters.vmSize }} WINDOWS_OS_SKU=${{ parameters.windowsOsSku }} WINDOWS_VM_SKU=${{ parameters.vmSize }} WINDOWS_USERNAME=${WINDOWS_USERNAME} WINDOWS_PASSWORD=${WINDOWS_PASSWORD}
```

Comment:
to mantain consistency across variables, I think we should name `DUALSTCK_OVERLAY_SUB` to `SUB_AZURE_NETWORK_AGENT_BUILD_VALIDATIONS`